src/lsqecc/pauli_rotations/rotation.py

```diff
@@ -91,12 +91,14 @@ def __init__(self, no_of_qubit: int):
         self.qubit_num: int = no_of_qubit
         self.ops_list: List[PauliOperator] = [PauliOperator("I") for i in range(no_of_qubit)]
 
+    @abstractmethod
     def __str__(self) -> str:
         pass
 
     def __repr__(self) -> str:
         return str(self)
 
+    @abstractmethod
```

Comment:
Yep this is 100% the right kind of abstract method, I also missed it before it seems